### DISCUSSION TITLE
Dynamic Expression Building

### DISCUSSION THREAD

**Discussion by tricaricom** [serverpod/serverpod#2661] [Q&A] [answered]
I am trying to build a where clause where sometimes fields are included and sometimes they are not.  

For example, I have 3 fields in my database table: field1, field2, field3.   In one scenario, I can build a where like this:

```
final list = await MyTable.db.find(
          session,
          where: (t) => t.field1.inSet(field1Set) & t.field2.inSet(field2Set) & t.field3.inSet(field3Set)
);
```

But sometimes, the set for field2 does not exist, so I do not want to include it in the where:

```
final list = await MyTable.db.find(
          session,
          where: (t) => t.field1.inSet(field1Set) & t.field3.inSet(field3Set)
);
```

How can I build an Expression object dynamically if field2 has an empty set?  I see the WhereExpressionBuilder in the code but it's not clear how I can build an Expression that I can pass to the where, like this:

```
Expression whereClause = WhereExpressionBuilder(only include field1 and field3);

final list = await MyTable.db.find(
          session,
          where: (t) => whereClause
);
```

Thanks!

**Comment by Isakdl** [comment, marked as answer]
You can easily build the query in anyway you like just like you could combine any other conditions in normal dart code.

Example:
```dart
final list = await MyTable.db.find(
          session,
          where: (t) {
            var filter = t.field1.inSet(field1Set) & t.field3.inSet(field3Set);
            if(someCondition) {
              filter = filter & t.field2.inSet(field2Set);
            }
            return filter;
          }
);


```

**Reply by tricaricom** [reply]
Thanks!
